sdk/src/azure/core/az_base64.c

```diff
@@ -370,17 +376,40 @@ AZ_NODISCARD int32_t az_base64_get_max_encoded_size(int32_t source_bytes_size)
   return (((source_bytes_size + 2) / 3) * 4);
 }
 
-static AZ_NODISCARD int32_t _az_base64_decode(uint8_t* encoded_bytes)
+static int32_t _get_base64_decoded_char(int32_t index, _az_base64_mode mode)
+{
+  if (mode == _az_base64_mode_url)
+  {
+    if (index == 43 || index == 47)
+    {
+      return -1; // can't use + or / with URL encoding
+    }
+
+    if (index == 45)
+    {
+      index = 43; // - becomes a +
+    }
+    else if (index == 95)
+    {
+      index = 47; // _ becomes a /
+    }
+  }
+
+  return _az_base64_decode_array[index];
```

Comment:
Can we get rid of the array? it is 256 bytes, and it is not very readable (bunch of `-1`s). Can we replace it with an (inline?) function doing something like this?
```c
if (c >= 'A' && c <= 'Z')
{
  return (c - 'A');
}

if (c >= 'a' && c <= 'z')
{
  return 26 + (c - 'a');
}

if (c >= '0' && c <= '9')
{
  return 52 + (c - '0');
}

if (c == '+')
{
  return 62;
}

if (c == '/')
{
  return 63;
}

return -1;
```

Reply:
Yea I can do it in another PR